routers/api/packages/npm/npm.go

```diff
@@ -160,6 +163,22 @@ func UploadPackage(ctx *context.Context) {
 		}
 	}
 
+	repo, err := repo_model.GetRepositoryByURL(npmPackage.Metadata.RepositoryURL)
+	if err == nil {
+		flag := repo.OwnerID == ctx.Doer.ID
+
+		if !flag {
+			perms, err := access_model.GetUserRepoPermission(ctx, repo, ctx.Doer)
+			flag = err == nil && perms.CanWrite(unit.TypePackages)
+		}
+
+		if flag {
+			if err := packages_model.SetRepositoryLink(ctx, pv.PackageID, repo.ID); err != nil {
+				ctx.ServerError("SetRepositoryLink", err)
+			}
+		}
```

Comment:
When there's no permission, ensure to return a Not authorized/Forbidden status

Reply:
Thanks for the reminder! Have also moved the permission check before the handling of the upload to avoid confusion when publishing.

Reply:
Is this resolved?